test/src/base/MooseTestApp.C

```diff
@@ -282,16 +282,27 @@ InputParameters
 validParams<MooseTestApp>()
 {
   InputParameters params = validParams<MooseApp>();
+  /* MooseTestApp is special because it will have its own
+   * binary and we want the default to allow test objects.
+   */
+  params.addCommandLineParam<bool>(
```

Comment:
Is it possible to use the suppressParam() method so that `allow_test_objects` doesn't appear in moose_test_app? I suspect not because these command line params probably don't support quite the same interface. Still it would be slightly cleaner if that worked.